storage/engine/rocksdb/db.cc

```diff
@@ -39,33 +39,84 @@
 extern "C" {
 #include "_cgo_export.h"
 
-struct DBBatch {
+struct DBEngine {
+  rocksdb::DB* const rep;
+
+  DBEngine(rocksdb::DB* r)
+      : rep(r) {
+  }
+  virtual ~DBEngine() { }
+
+  virtual DBStatus Put(DBSlice key, DBSlice value);
+  virtual DBStatus Merge(DBSlice key, DBSlice value);
+  virtual DBStatus Delete(DBSlice key);
+  virtual DBStatus WriteBatch();
+  virtual DBStatus Get(DBSlice key, DBString* value);
+  virtual DBIterator* NewIter();
+};
+
+struct DBImpl : public DBEngine {
+  rocksdb::Env* const memenv;
+
+  DBImpl(rocksdb::DB* r, rocksdb::Env* m)
+      : DBEngine(r),
+        memenv(m) {
+  }
+  virtual ~DBImpl() {
+    delete rep;
+    delete memenv;
```

Comment:
We require c++11 in `c-rocksdb/cgo_flags.go` and no one has complained yet, so I think we'd be fine with `unique_ptr`.